src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_cthun.cpp

```diff
@@ -585,6 +599,43 @@ struct boss_cthun : public BossAI
 
         //Body Phase
         uint8 _fleshTentaclesKilled;
+
+        //If no-one is outside
+        uint32 _acidTimer;
+
+        bool IsEveryoneAffectedByDigestiveAcid() const
+        {
+            Map* map = me->GetMap();
+            if (!map->IsDungeon())
+                return false;
+
+            Map::PlayerList const& playerList = map->GetPlayers();
+            for (const auto& itr : playerList)
+            {
+                if (Player* player = itr.GetSource())
+                {
+                    if (!player->HasAura(SPELL_DIGESTIVE_ACID))
+                        return false;
+                }
+            }
+            return true;
+        }
+
+        void KillEveryone()
+        {
+            Map* map = me->GetMap();
+            if (!map->IsDungeon())
+                return;
+
+            Map::PlayerList const& playerList = map->GetPlayers();
+            for (const auto& itr : playerList)
+            {
+                if (Player* player = itr.GetSource())
+                {
+                    player->KillSelf();
```

Comment:
On classic/live, does the player in fact kill themselves or is there a spell in play here that needs to be cast by C'Thun?

Reply:
It likely is a spell and it would be worthwhile to try and get a sniff for it

Reply:
> It likely is a spell and it would be worthwhile to try and get a sniff for it

I have no sniff to give, but it does seem CMaNGOS has this one figured out. Specifically, they make use of [Port Out Stomach Effect](https://wowgaming.altervista.org/aowow/?spell=26648). (Can be seen [here](https://github.com/cmangos/mangos-wotlk/blob/master/src/game/AI/ScriptDevAI/scripts/kalimdor/temple_of_ahnqiraj/boss_cthun.cpp#L78))